### PR TITLE
Support regenerating Query API Key

### PR DESCRIPTION
**Pull request by kyoshidajp**
## What type of PR is this? (check all applicable)

- [x] Feature

## Description

Support regenerating Query API Key.

Add API endpoint to regenerate.

```
POST /api/queries/{id}/regenerate_api_key
```

Also its button (ref: the screenshot below at the bottom of this page).

## Related Tickets & Documents

- [Regenerate API KEY \- Feature Requests \- Redash Discourse](https://discuss.redash.io/t/regenerate-api-key/2134/2)

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

Query API Key dialog:
after_567x335

### CODE REVIEW

File: client/app/components/queries/api-key-dialog.js [hunk 1]
```diff
@@ -4,26 +4,46 @@ const ApiKeyDialog = {
...
+    $scope.query = this.resolve.query;
+    this.csvUrlBase = `${clientConfig.basePath}api/queries/${this.resolve.query.id}/results.csv?api_key=`;
+    this.jsonUrlBase = `${clientConfig.basePath}api/queries/${this.resolve.query.id}/results.json?api_key=`;
+
+    this.regenerateQueryApiKey = () => {
+      $scope.disableRegenerateApiKeyButton = true;
+      $http
+        .post(`api/queries/${this.resolve.query.id}/regenerate_api_key`)
+        .success((data) => {
+          $scope.query = data;
```
Comment (arikfr) [inline client/app/components/queries/api-key-dialog.js]: See my comment about simplifying the response of this API call. In this case, we only need to update the API key of the query object.

File: client/app/components/queries/api-key-dialog.js [hunk 2]
```diff
@@ -4,26 +4,46 @@ const ApiKeyDialog = {
...
+    this.api_key = this.resolve.query.api_key;
+    this.csvUrlBase = `${clientConfig.basePath}api/queries/${this.resolve.query.id}/results.csv?api_key=`;
+    this.jsonUrlBase = `${clientConfig.basePath}api/queries/${this.resolve.query.id}/results.json?api_key=`;
+
+    this.regenerateQueryApiKey = () => {
+      this.disableRegenerateApiKeyButton = true;
+      $http
+        .post(`api/queries/${this.resolve.query.id}/regenerate_api_key`)
+        .success((data) => {
+          this.api_key = data.api_key;
```
Comment (kyoshidajp) [inline client/app/components/queries/api-key-dialog.js]: 1. Regenerate API Key
1. Close dialog
1. Reopen dialog (API Key isn't updated)

Need to update the query object, but I have no idea.

Comment (arikfr) [reply client/app/components/queries/api-key-dialog.js]: You should update the `query.api_key` for it to persist.

Comment (kyoshidajp) [reply client/app/components/queries/api-key-dialog.js]: Thanks. Fixed it.

**Comment by kyoshidajp** [comment]
@kravets-levko Thanks. Please review again.

**Comment by arikfr** [comment]
Thanks!

**Pull request merged**